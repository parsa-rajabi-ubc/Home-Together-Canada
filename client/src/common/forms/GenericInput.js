/**
 * @Author:     Parsa Rajabi
 * @Created:    2020.11.19
 *
 * @Description: Generic Input
 *
 */

import React from 'react';
import PropTypes from "prop-types";

function GenericInput(props) {
    const {className, inputType, placeholder, onChange} = props;
    return (

        <label>
            {label}
            <input className={className} type={inputType} placeholder={placeholder} onChange={onChange}/>
        </label>
    );
}

GenericInput.propTypes = {
    label: PropTypes.string.isRequired,
    inputType: PropTypes.string.isRequired,
    className: PropTypes.string,
    placeholder: PropTypes.string,
    onChange: PropTypes.func
};

export default GenericInput;