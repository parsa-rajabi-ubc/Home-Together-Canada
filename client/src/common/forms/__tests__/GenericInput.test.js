/**
 * @Author:     Parsa Rajabi
 * @Created:    2020.11.19
 *
 * @Description: Generic input Component Snapshot test
 *
 */

import React from 'react';
import GenericInput from "../GenericInput";
import renderer from  'react-test-renderer'

describe('GenericInput', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties", () => {
            // given
            const label = 'testLabelString1';
            const type = 'testTypeString1';
            const className = 'testClassNameString';
            const placeholder = 'testPlaceholderString2';

            // when
            const component = renderer.create(<GenericInput className={className} inputType={type} label={label} placeholder={placeholder}/>).toJSON();

            // then
            expect(component).toMatchSnapshot();
        });
    })
})