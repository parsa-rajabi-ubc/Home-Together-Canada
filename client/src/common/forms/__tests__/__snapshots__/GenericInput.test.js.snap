// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericInput Snapshot test should render correctly regardless of properties 1`] = `
<label>
  testLabelString1
  <input
    className="testClassNameString"
    placeholder="testPlaceholderString2"
    type="testTypeString1"
  />
</label>
`;
